base/sysimg.jl

```diff
@@ -96,7 +96,6 @@ include("subarray.jl")
 (::Type{Vector})() = Array{Any,1}(0)
```

Comment:
I'd also favor deprecating `Vector()` and `Vector{T}()`, but doing so in a separate PR due to the expected controversy sounds like a good idea.